pkg/deploy/sync.go

```diff
@@ -9,157 +9,249 @@ import (
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 // Sync syncs the blueprint to the cluster in a generic (as much as Go allows) manner.
-// Returns true if the object was created or updated, false if there was no change detected.
+// Returns true if object is up to date otherwiser returns false
 func Sync(deployContext *DeployContext, blueprint metav1.Object, diffOpts cmp.Option) (bool, error) {
-	key := client.ObjectKey{Name: blueprint.GetName(), Namespace: blueprint.GetNamespace()}
-
-	actual, err := Get(deployContext, key, blueprint)
-	if err != nil {
-		return false, err
+	if !deployContext.CheCluster.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
Maybe we should apply comment here and explain, that we returns true for fine finalization. Or update doc comment for 'Sync' method and apply this info there.